streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamsKafkaClientTest.java

```diff
@@ -124,6 +131,78 @@ public void shouldNotAllowNullTopicConfigs() throws Exception {
         verifyCorrectTopicConfigs(streamsKafkaClient, topicConfigWithNoOverrides, Collections.singletonMap("cleanup.policy", "delete"));
     }
 
+    @Test
+    public void testCheckBrokerCompatibilityEmptyResponse() {
+        kafkaClient.prepareResponse(null);
+        final StreamsKafkaClient streamsKafkaClient = createStreamsKafkaClient();
+        try {
+            streamsKafkaClient.checkBrokerCompatibility(false);
+            fail("Response was not empty");
+        } catch (final StreamsException se) {
+            assertThat(se.getMessage(), equalTo("Empty response for client request."));
+        }
+    }
+
+    @Test
+    public void testCheckBrokerCompatibilityConsistentResponseType() {
+        kafkaClient.prepareResponse(new ProduceResponse(Collections.<TopicPartition, ProduceResponse.PartitionResponse>emptyMap()));
+        final StreamsKafkaClient streamsKafkaClient = createStreamsKafkaClient();
+        try {
+            streamsKafkaClient.checkBrokerCompatibility(false);
+            fail("Consistent response type");
+        } catch (final StreamsException se) {
+            assertThat(se.getMessage(), equalTo("Inconsistent response type for API versions request. Expected ApiVersionsResponse but received org.apache.kafka.common.requests.ProduceResponse"));
+        }
+    }
+
+    @Test
+    public void testCheckBrokerCompatibilityRequiresBrokerVersion0101xOrHigher() {
```

Comment:
as above. Name of the test can be more descriptive. `shouldRequireBrokerVersion0101OrHigherWhenEosDisabled`